### DISCUSSION TITLE
[GKE] [Cloud/managed K8S] Scaphandre pod is always pending in Kubernetes cluster

### DISCUSSION THREAD

**Discussion by annis-souames** [hubblo-org/scaphandre#277] [Q&A] [answered]
## Bug description

I tried to install Scaphandre on a basic Kubernetes cluster hosted on GKE but the Scaphandre node have been pending for more than 24 hours since installation. The status of the pods shown by `kubectl` is `ContainerCreating` 

## To Reproduce

1 - Setup a basic Kubernetes cluster on GKE
2 - Install Scaphandre, Grafana and Prometheus following the [Scahandre on Kubernetes documentation](https://hubblo-org.github.io/scaphandre-documentation/tutorials/kubernetes.html). 

## Expected behavior

The pods should be running like Grafana and Prometheus related pods

## Environment

- Kubernetes 1.25
- Latest version of Scaphandre (installed it yesterday using helm (0.5.0)) 

## Additional context

Also, no metrics `scaph_` is showing up in the metrics explorer on prometheus and Grafana dashboard for Scaphander (loaded using the provided template json) is empty.

**Comment by demeringo** [comment, marked as answer]
Hello, thanks for opening the issue.

I believe that GKE (and other flavor of public cloud provider managed kubernetes) may not provide access to RAPL.
This may explain the lack of metrics.

I believe you need to deploy Scaphandre to a physical/bare metal server (like an on premise kubernetes cluster) that do not restrict access to RAPL.

Maybe this explains why the pod does not starts up well  (although in my opinion, it _should_ start but still report empty of zero metrics).

**Comment by annis-souames** [comment]
Thank you, that seems the problem indeed, GCP does not give access to the RAPL metrics. Do you know any other way to get the power consumption in a managed cloud without relying on RAPL ?

**Comment by demeringo** [comment]
Hi, 
Sadly there is no tool - yet - that fully allow to monitor/measure power consumption on public cloud platform.
We can estimate (using a model) power usage of instances through.
Best tools I know in this space are 
- Boavizta cloud-scanner (only supports _aws_ ec2 instances): https://github.com/Boavizta/cloud-scanner
- Cloud Carbon footprint : https://github.com/cloud-carbon-footprint/

**Comment by demeringo** [comment]
Hi @bpetit, maybe we could close the issue (or even better move it to a discussion in github ?).

**Comment by bpetit** [comment]
Also to mention https://github.com/Boavizta/Energizta/ that should allow to build accurate models to do so at some point !

Yes @demeringo converting as a discussion is a good idea ! Thanks
